aten/src/ATen/native/cuda/BinaryRemainderKernel.cu

```diff
@@ -3,12 +3,18 @@
 #include <ATen/native/cuda/Loops.cuh>
 #include <ATen/native/TensorIterator.h>
 #include <ATen/native/BinaryOps.h>
+#include <ATen/native/cuda/Math.cuh>
 
 // NOTE: CUDA on Windows requires that the enclosing function
 // of a __device__ lambda not have internal linkage.
 
 namespace at { namespace native {
 
+template<typename T>
+__host__ __device__ static inline c10::complex<T> floor_wrapper(c10::complex<T> v) {
+  return c10::complex<T>(std::floor(v.real()), std::floor(v.imag()));
+}
+
```

Comment:
I am not sure this is the right way to implement `floor` for complex. I would go for
```
floor(z)  = ceil(|z|) * z/|z|.
```

Reply:
The reason being is that I would assume that `floor(z)` and `z` should have the same Argument (angular information). And taking floor in imaginary and real parts separately kind of breaks the argument.